modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java

```diff
@@ -59,6 +66,11 @@ public class ScaleCubeClusterServiceFactory {
     /** Logger. */
     private static final IgniteLogger LOG = Loggers.forClass(ScaleCubeClusterServiceFactory.class);
 
+    /** Metadata codec. */
+    private static final NodeMetadataCodec METADATA_CODEC = NodeMetadataCodec.INSTANCE;
+    /** Scalecube cluster executor to update metadata.  */
+    private final Scheduler scheduler = Schedulers.fromExecutor(Executors.newSingleThreadExecutor());
```

Comment:
Why do you need this executor?

Reply:
Because we should provide an executor when calling io.scalecube.cluster.ClusterImpl#updateMetadata. Without that metadata will not be updated ever.

Reply:
Why? I can see that `updateMetadata` is dispatched on a ScaleCube's scheduler internally:
```
return Mono.fromRunnable(() -> metadataStore.updateMetadata(metadata))
        .then(membership.updateIncarnation())
        .subscribeOn(scheduler);
```

Reply:
done